driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java

```diff
@@ -128,6 +134,19 @@ public void makeActive() {
         }
     }
 
+    public void makeActive() {
+        makeActiveDb();
+
+        try {
+            this.database.getMetadata().getSchema().reload();
+        } catch (Exception _) {
+            this.database = new ODatabaseDocumentTx(this.database.getURL(), this.database.isKeepStorageOpen());
```

Comment:
You are right but there are 2 problems, 
1. the database may have not been created using a pool (poll may be null)
2. `pool.acquire();` gives me a database with a failing connection (if there was a problem in the first place), so i really need to recreate the it by hand.

One performance change will follow.